core/src/forest/ForestOptions.cpp

```diff
@@ -62,6 +62,31 @@ ForestOptions::ForestOptions(uint num_trees,
   }
 }
 
+ForestOptions::ForestOptions(uint num_trees,
+                             size_t ci_group_size,
+                             double sample_fraction,
+                             uint mtry,
+                             uint min_node_size,
+                             bool honesty,
+                             double honesty_fraction,
+                             bool honesty_prune_leaves,
+                             double alpha,
+                             double imbalance_penalty,
+                             uint num_threads,
+                             uint random_seed,
+                             const std::vector<size_t>& sample_clusters,
+                             uint samples_per_cluster,
+                             uint blocklength) :
+    ci_group_size(ci_group_size),
+    sample_fraction(sample_fraction),
+    tree_options(mtry, min_node_size, honesty, honesty_fraction, honesty_prune_leaves, alpha, imbalance_penalty),
+    sampling_options(samples_per_cluster, sample_clusters, blocklength){
+  
+    ForestOptions(num_trees, ci_group_size, sample_fraction, mtry, min_node_size,
```

Comment:
The constructor need to be after the colon. e.g.,
`class Foo {
public: 
  Foo(char x, int y) {}
  Foo(int y) : Foo('a', y) {} // Foo(int) delegates to Foo(char,int)
};`